src/collection/collectionmodel.cpp

```diff
@@ -165,6 +170,19 @@ void CollectionModel::SaveGrouping(QString name) {
 
 }
 
+void CollectionModel::ReloadSettings() {
+
+  if (UseDiskCache()) {
```

Comment:
I assume you meant !UseDiskCache() here? We want to clear the cache when it's disabled in the setting, otherwise the cache will be cleared each time the settings is saved.